src/cli/index.js

```diff
@@ -142,7 +142,7 @@ if (outputWrapper) {
 if (command.noArguments && commander.args.length) {
   reporter.error(reporter.lang('noArguments'));
   reporter.info(command.getDocsInfo);
-  process.exit(1);
+  process.exitCode = 1;
```

Comment:
I don't think that's a good idea.

- it kinda violates the principle of the least surprise, since we would no longer have the guarantee that any part of this file would only be executed if no other command matched. It's only a matter of time before we make some mistake that will execute two commands in some edge case.

- Case in point, your current code has a bug that will make `yarn -v` print the version number before running `yarn install -v` (and `yarn -v -h` will print the version number then call `yarn help -v`).

- Even without that, not shortcutting also means that we will be running useless tasks (such as instanciating a reporter etc) even when not needed. That's an issue because those commands must be fast (some tools such as Nuclide might need to call them often, and if their results are late it will slow down the all process).

I'm not against using `exitCode`, you have a point that shutting down the program before the I/O has finished is bad, but I strongly think we then need to refactor the CLI code to make sure we never have to check that our code won't get called twice - might only be a matter of adding `else` clauses 😃